lib/compress/zstdmt_compress.c

```diff
@@ -361,76 +364,70 @@ void ZSTDMT_compressChunk(void* jobDescription)
                 goto _endJob;
         }   }
         {   size_t const initError = ZSTD_compressBegin_advanced_internal(cctx,
-                                        job->srcStart, job->prefixSize, ZSTD_dm_rawContent, /* load dictionary in "content-only" mode (no header analysis) */
+                                        job->prefixStart, job->prefixSize, ZSTD_dm_rawContent, /* load dictionary in "content-only" mode (no header analysis) */
                                         NULL, /*cdict*/
                                         jobParams, pledgedSrcSize);
             if (ZSTD_isError(initError)) {
                 job->cSize = initError;
                 goto _endJob;
-        }   }
-    }
+    }   }   }
     if (!job->firstChunk) {  /* flush and overwrite frame header when it's not first job */
-        size_t const hSize = ZSTD_compressContinue(cctx, dstBuff.start, dstBuff.size, src, 0);
+        size_t const hSize = ZSTD_compressContinue(cctx, dstBuff.start, dstBuff.capacity, src, 0);
         if (ZSTD_isError(hSize)) { job->cSize = hSize; /* save error code */ goto _endJob; }
         DEBUGLOG(5, "ZSTDMT_compressChunk: flush and overwrite %u bytes of frame header (not first chunk)", (U32)hSize);
         ZSTD_invalidateRepCodes(cctx);
     }
 
     /* compress */
-#if 0
-    job->cSize = (job->lastChunk) ?
-                 ZSTD_compressEnd     (cctx, dstBuff.start, dstBuff.size, src, job->srcSize) :
-                 ZSTD_compressContinue(cctx, dstBuff.start, dstBuff.size, src, job->srcSize);
-#else
-    if (sizeof(size_t) > sizeof(int)) assert(job->srcSize < ((size_t)INT_MAX) * ZSTD_BLOCKSIZE_MAX);   /* check overflow */
-    {   int const nbBlocks = (int)((job->srcSize + (ZSTD_BLOCKSIZE_MAX-1)) / ZSTD_BLOCKSIZE_MAX);
+    {   size_t const blockSize = ZSTD_BLOCKSIZE_MAX;
```

Comment:
Is it possible for the user to pass in a window size < `ZSTD_BLOCKSIZE_MAX`?

Reply:
Let's see,
A `windowSize < ZSTD_BLOCKSIZE_MAX` presumes a `srcSize < ZSTD_BLOCKSIZE_MAX`.
In such situation, when `windowSize` is automatically set by `zstd`, it always ensures that `windowSize >= srcSize`. So the algorithm skips the loop and goes straight to "last block".

However, we could be in a less common situation,
where `windowSize` has been set manually to a low value, 
while `srcSize` could be very large.
In which case, the real `blockSize` used in the frame will be `== windowSize < ZSTD_BLOCKSIZE_MAX`,
while `ZSTDMT_compressChunk()` will continue to feed `ZSTD_compressContinue()` with inputs of `ZSTD_BLOCKSIZE_MAX` bytes.

That being, said, it's not necessarily a problem.
`ZSTD_compressContinue()` will, internally, cut this input into as many blocks as necessary.
It knows the size of blocks : it's stored as `cctx->blockSize`, and was set during initialisation.
`cctx->blockSize` has taken into consideration `windowSize`.

All is well as long as `blockSize` is a clean power of 2, so that `inputSize == ZSTD_BLOCKSIZE_MAX` will be divided into an integer number of blocks. The specification allows non-power-of-2 window sizes, and by extension, block sizes, but nothing in the code make it possible right now.
Besides, in such a case, the drawback will be that `ZSTD_compressContinue()` will create a "small last block", but that's a minor setback, everything will continue to work and respect the specification.

I guess I'd better rename this variable, so that it's not confused with internal block size.